### PR TITLE
fix: correctly trim resolved IPNS addresses

### PR DESCRIPTION
**Pull request by Stebalien**
fixes https://github.com/ipfs/go-ipfs/issues/7246

### CODE REVIEW

**Comment by Stebalien** [comment]
This test is tests resolving keys through the `/ipns` mountpoint. That code just calls `Resolve` using the CoreAPI so the test should be generic.

We do need to start running this in CI, but we'll need a special job for fuse-only test.

> Does it work successfully for you?

It does... do the other tests work?

File: test/sharness/t0030-mount.sh [hunk 1]
```diff
@@ -53,6 +55,12 @@ test_expect_success FUSE "'ipfs mount' output looks good" '
   test_cmp expected actual
 '
 
+test_expect_success FUSE "can resolve ipns names" '
```
Comment (aschmahmann) [inline test/sharness/t0030-mount.sh:58]: Continuing from https://github.com/ipfs/go-ipfs/pull/7331#issuecomment-631180756 so we can do this in a thread-like fashion.

Running `./t0030-mount.sh -v` passes on master, but with this branch fails with:
`cat: ipns/welcome.example.com/ping: No such file or directory`

Comment (Stebalien) [reply test/sharness/t0030-mount.sh:58]: Did you make sure to rebuild? Otherwise, I have no idea. Does `ipns/local` exist when running the test?

Comment (aschmahmann) [reply test/sharness/t0030-mount.sh:58]: I think I had a rebuilding issue... so dumb. Sorry for the confusion

**Pull request merged**